### PR TITLE
Add initial CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by bjosv**
Move the contributing headline from `README.md` to own file, as described in the [organizations README.md](https://github.com/valkey-io#contributing).
Add DCO information and some initial/minimal coding conventions.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
+# Contributing
+
+:tada:Thanks for taking the time to contribute!:tada:
```
Comment (zuiderkwast) [inline CONTRIBUTING.md:3]: Some space?

```suggestion
:tada: Thanks for taking the time to contribute! :tada:
```

File: README.md [hunk 2]
```diff
@@ -65,6 +65,4 @@ sudo make install
 
 ## Contributing
 
-Contributions from the community are welcome. Feel free to open a PR with a bugfix or new feature. Note that if you would like to merge a large feature it's probably a good idea to open an issue first rather than spending a lot of time on a PR that may not be accepted.
-
-When making a change, please use `git clang-format` or [format-files.sh](./scripts/format-files.sh) to format your changes properly. This repository is currently using `clang-format` 18.1.3 to format the code.
+Please see the [`CONTRIBUTING.md`](https://github.com/valkey-io/libvalkey/blob/main/CONTRIBUTING.md).
```
Comment (zuiderkwast) [inline README.md:68]:
```suggestion
Please see [`CONTRIBUTING.md`](https://github.com/valkey-io/libvalkey/blob/main/CONTRIBUTING.md).
```

.... or *"the `CONTRIBUTING.md` file"*.

**Review by zuiderkwast — APPROVED** [review]
Great!

Just a few nits.

File: README.md (hunk 2 above)
Comment (bjosv) [reply README.md:68]: Thanks!

**Pull request merged**